### DISCUSSION TITLE
Is switching to the Free Spaced Repetition Scheduler a worthwhile improvement?

### DISCUSSION THREAD

**Discussion by TomRomeo** [kioku-project/kioku#277] [Ideas]
Taken from their own [Github page](https://github.com/open-spaced-repetition), Open Spaced Repetition is ...
> [...] an open-source community dedicated to the advancement of spaced repetition algorithms. We are the creators of the Free Spaced Repetition Scheduler (FSRS), which is backed by academic literatures and experiments.

Their own Spaced Repetition Scheduler (FSRS) can also be used in the new versions of Anki and is the result of several academic papers. From their own benchmarks, it seems to perform comparable to the newer SM-17 SuperMemo algorithms. [[1]](https://github.com/open-spaced-repetition/srs-benchmark?tab=readme-ov-file), [[2]](https://github.com/open-spaced-repetition/fsrs-vs-sm17)

Since our scheduler is very basic, it might be a good idea to replace the current implementation with one that uses the FSRS, especially now while we rewrite the card logic and database schemas.

Of course, this option should be thoroughly investigated. I will list some Pros and Cons that came to my mind below and would love to hear your opinions and feedback!

### Pros
- Even better compatibility with Anki - Using the same algorithm as Anki would allow us to offer even more compatibility with Anki as users will find both algorithms to behave similarly and we could convert a users Anki learning progress into Kioku as well when they upload their decks
- Improved Spaced Repetition Algorithm - Compared to our current implementation, the FSRS would probably lead to better results as it is way more optimized
- We can focus on other parts of the application - If we implement the FSRS algorithm, we will have to spend less time on advancing and optimizing our own SRS and hopefully have more time for other features

### Cons
- potentially harder to modify to our needs - if we want to diverge from the implementation, we could of course fork a current implementation and make changes if necessary, however this would potentially lead to a lot of additional work and maintenance
- ?

**Comment by memeToasty** [comment]
I think it would be a great idea to move to FSRS, because I don't think that we should modify a lot in regards to the scheduling.

**Comment by 4KevR** [comment]
I also think that moving to FSRS would be a good idea. Of course there is the point of minimal "lock-in" but I would say that it is more beneficial for the project to have a proven algorithm for the SRS. Also the fact that Anki users can have a "familiar" experience would be beneficial. I think the main part of moving would include the adjustment of our rating system as FSRS expects 4 categories. But I think I like the idea of including an explicit "again" category also to our system.
